schema.graphql

```diff
@@ -2329,3 +2358,154 @@ type CustomClaimType @entity {
   createdBlock: Block!
   updatedBlock: Block!
 }
+
+enum AffirmingPartyEnum {
+  Sender
+  Receiver
+  Mediator
+}
+
+enum AffirmStatusEnum {
+  Affirmed
+  Rejected
+}
+
+enum ConfidentialTransactionStatusEnum {
+  Created
+  Executed
+  Rejected
+}
+
+"""
+Represents the proof of a transaction affirmation
+"""
+type SenderProof @jsonField {
+  assetId: String!
+  proof: String!
+}
+
+"""
+Represents a confidential account
+"""
+type ConfidentialAccount @entity {
+  id: ID!
+  account: String!
+  creator: Identity!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential venue
+"""
+type ConfidentialVenue @entity {
+  id: ID!
+  venueId: Int!
+  creator: Identity!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential asset
+"""
+type ConfidentialAsset @entity {
+  id: ID!
+  assetId: String!
+  ticker: String
+  data: String
+  creator: Identity!
+  auditors: [String!]
+  mediators: [String!]
+  totalSupply: BigInt!
+  allowedVenues: [Int!]
+  venueFiltering: Boolean!
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential asset holder
+"""
+type ConfidentialAssetHolder @entity {
+  id: ID!
+  account: ConfidentialAccount!
+  asset: ConfidentialAsset!
+  """
+  the encrypted amount
+  """
+  amount: String
+  createdBlock: Block!
+  updatedBlock: Block!
+}
+
+"""
+Represents a confidential asset transaction affirmation
+"""
+type ConfidentialTransactionAffirmation @entity {
+  id: ID!
+  transaction: ConfidentialTransaction!
+  leg: Int!
+  account: ConfidentialAccount
+  identity: Identity!
+  type: AffirmingPartyEnum!
+  status: AffirmStatusEnum!
+  proofs: [SenderProof]
```

Comment:
```suggestion
  proofs: [SenderProof!]
```
The array is optional, but if its present, then the element isn't. Although it could still be empty

I am not sure, but that would be consistent with `[AssetAuditorGroup!]` below.